### PR TITLE
epic(website): refactor website

### PR DESCRIPTION
**Pull request by jw-foss**
Please make sure these boxes are checked before submitting your PR, thank you!

# THIS IS THE EPIC BRANCH FOR REFACTORING THE WEBSITE

- Use VitePress as the framework to contain the website
- Use Crowdin as the translation service for i18n
- Redesigned the entire website UI.

## Credits

In this version we have adapted a lot of layout and UX ideas from [@vue/theme](https://github.com/vuejs/docs/tree/next). :pray: 

## Future plan

We are preparing dark mode (like Vue Docs did) for users (also dark mode theme for ElementPlus itself).

- [ ] Make sure you follow Element's contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: .github/workflows/staging-preview.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+name: Staging Preview
+
+on:
+  push:
+    branches:
+      - 'dev'
+      - 'epic/website-refactor'
```
Comment (adaex) [inline .github/workflows/staging-preview.yml:7]: ❕ Note: Be sure to delete this line before merging to dev

File: .gitignore [hunk 2]
```diff
@@ -14,3 +14,16 @@ lib
 packages/element-plus/version.ts
 /es/
 es
+
+.vitepress/*.js
```
Comment (sxzz) [inline .gitignore:18]: IMO, I think these should be moved to `docs/.gitignore`

File: README.md [hunk 3]
```diff
@@ -12,11 +12,19 @@
...
+- 💪 Vue 3 Composition API
 - 🔥 Written in TypeScript
 
+## Archived website
+
+If you are looking for previous version website, here is the link.
+
+[Element Plus Documentation Archived](https://github.com/element-plus/doc-archive)
+
+The new website is launched at 15th Sep 2021.
```
Comment (sxzz) [inline README.md:26]: update date once merged.

**Review by adaex — APPROVED** [review]

File: .gitignore (hunk 2 above)
Comment (JeremyWuuuuu) [reply .gitignore:18]: You were right, let's move it

**Review by iamkun — APPROVED** [review]

**Review by sxzz — APPROVED** [review]

**Comment by JeremyWuuuuu** [comment]
This is a big milestone for all of us!

**Pull request merged**